### ISSUE TITLE
Problem with requests

### ISSUE THREAD

**Issue by teawithfruit** [slact/nchan#610]
Hi. Thank you for developing this lib. I like the idea of putting all the connection types together to a single library.
At this point I'am trying to configure nginx to handle all this live connection stuff with nchan and I ran into some problems.
This is my current config:

```nginx
location /nchan_stub_status {
  nchan_stub_status;
}

location = /auth {
  proxy_pass http://live/auth;
  proxy_pass_request_body off;
  proxy_set_header Content-Length "";
  proxy_set_header X-Subscriber-Type $nchan_subscriber_type;
  proxy_set_header X-Publisher-Type $nchan_publisher_type;
  proxy_set_header X-Prev-Message-Id $nchan_prev_message_id;
  proxy_set_header X-Channel-Id $nchan_channel_id;
  proxy_set_header X-Original-URI $request_uri;
  proxy_set_header X-Forwarded-For $remote_addr;
}

location = /pub {
  proxy_pass http://live/pub;
  proxy_set_header X-Publisher-Type $nchan_publisher_type;
  proxy_set_header X-Prev-Message-Id $nchan_prev_message_id;
  proxy_set_header X-Channel-Id $nchan_channel_id;
  proxy_set_header X-Original-URI $request_uri;
}

location = /sub {
  proxy_pass http://live/sub;
  proxy_set_header X-Subscriber-Type $nchan_subscriber_type;
  proxy_set_header X-Message-Id $nchan_message_id;
  proxy_set_header X-Channel-Id $nchan_channel_id;
  proxy_set_header X-Original-URI $request_uri;
}

location = /unsub {
  proxy_pass http://live/unsub;
  proxy_ignore_client_abort on;
  proxy_set_header X-Subscriber-Type $nchan_subscriber_type;
  proxy_set_header X-Channel-Id $nchan_channel_id;
  proxy_set_header X-Original-URI $request_uri;
}

location /p {
  nchan_channel_id $arg_channel;
  nchan_publisher;
  nchan_redis_pass redis;

  nchan_publisher_upstream_request /pub;
  #nchan_authorize_request /auth;
}

location /s {
  nchan_channel_id $arg_channel;
  nchan_subscriber;
  nchan_redis_pass redis;
  nchan_channel_id_split_delimiter ",";
  nchan_subscriber_first_message newest;
  nchan_websocket_ping_interval 20;
  
  #nchan_authorize_request /auth;
  #nchan_subscribe_request /sub;
  #nchan_unsubscribe_request /unsub;
}

location /ps {
  nchan_channel_id $arg_channel;
  nchan_pubsub;
  nchan_redis_pass redis;
  nchan_channel_id_split_delimiter ",";
  nchan_subscriber_first_message newest;
  nchan_websocket_ping_interval 20;
  
  #nchan_publisher_upstream_request /pub;
  #nchan_authorize_request /auth;
  #nchan_subscribe_request /sub;
  #nchan_unsubscribe_request /unsub;
}
```

I've read the documentation and also many files in this repo, but I cannot figure out what I am doing wrong.
The problem is, that all requests excepting ```nchan_publisher_upstream_request``` in the ```/p``` route do not work. I've tried it in many ways. With and without REDIS, with and without all the other config lines but nginx gives me all the time the following error for the currently excepted request lines in ```/s``` and ```/ps```.
```
2021/05/11 09:09:15 [alert] 1#1: worker process 24 exited on signal 11
```

```nchan_authorize_request``` in ```/p``` works half way. I got a request on my server, but then the nginx get stuck without an error. The POST request to publish something just don't get closed.

Currently I'm using ```nginx 1.19.10```.

For testing the configuration I'm using wscat and curl with the following commands:
```
curl --request POST --data "test message" https://127.0.0.1/p?channel=abc
wscat -n -c wss://127.0.0.1/ps?channel=abc
wscat -n -c wss://127.0.0.1/s?channel=abc
```

Do you have any suggestions for me?
What am I doing something wrong?

Many thanks for your help!

**Comment by teawithfruit** [comment]
I figured out now, that this behavior starts with ```nginx 1.19.9```. So with version ```1.19.8```, the requests are working.
However, the problem with ```nchan_authorize_request``` unfortunately still exists.

It looks like nginx/nchan can't handle the response from the server, although there's only a 200 status code coming back at the moment.

**Comment by teawithfruit** [comment]
Now I have some new findings.
It's not about the ```nchan_authorize_request``` request alone.
This problem happens, if I use ```nchan_authorize_request``` and ```nchan_publisher_upstream_request``` together, like I've done it in the ```/p``` route.

For testing I use this CURL:
```
curl --request POST --data "TEST" https://127.0.0.1/p?channel=abc
```

If I remove the ```nchan_authorize_request``` request, everything works and the response looks something like this:
```json
{"messages": 7, "requested": 40, "subscribers": 43, "last_message_id": "1620841017:0" }
```

If I use it with ```nchan_authorize_request``` in combination, the message is also delivered, but CURL gives me the following response after 2 minutes with another copy of the message.
```
curl: (52) Empty reply from server
```

I don't know what the problem could be. Maybe something about the request order or is it just impossible to combine this two requests? Or is it just a problem with the non websocket behavior?

@slact Do you have any suggestions? Would appriciate your help. :)

**Comment by ghostmaster-ai** [comment]
@teawithfruit I tried using nginx module "auth_request" which is similar to "nchan_authorize_request" with http and websockets at subscriber location, it works just fine.
i am using:
ubuntu 20.04 LTS
nginx 1.20.1
nchan 1.2.12

**Comment by slact** [comment]
Nginx 1.19.8 and above broke `nchan_publisher_upstream_request` and `nchan_authorize_request`. This is now fixed in master, and will be released shortly in 1.2.14
